kie-server-parent/kie-server-tests/kie-server-integ-tests-all/src/test/java/org/kie/server/integrationtests/prometheus/PrometheusIntegrationTest.java

```diff
@@ -232,11 +232,9 @@ public void testPrometheusJobErrorMetrics() throws Exception {
 		Long jobId = jobServicesClient.scheduleRequest(jobRequestErrorInstanceNow);
 		KieServerSynchronization.waitForJobToFinish(jobServicesClient, jobId);
 
-		assertThat(getMetrics()).contains(
+		assertThat(getMetrics()).containsPattern(
 				"kie_server_job_in_retry_total{container_id=\"\",failed=\"true\",command_name=\""
-						+ JOB_EXECUTION_ERROR_COMMAND + "\",} 0.0",
-				"kie_server_job_error_total{container_id=\"\",failed=\"true\",command_name=\""
-						+ JOB_EXECUTION_ERROR_COMMAND + "\",} ");
```

Comment:
Why not to keep it in a different assert?

Reply:
because it is a flaky test. you cannot guarantee the race condition.